packages/pigeon/example/app/ios/Runner/EventChannelMessages.g.swift

```diff
@@ -23,6 +23,67 @@ private func nilOrValue<T>(_ value: Any?) -> T? {
   return value as! T?
 }
 
+func deepEqualsEventChannelMessages(_ lhs: Any?, _ rhs: Any?) -> Bool {
+  let cleanLhs = nilOrValue(lhs) as Any?
+  let cleanRhs = nilOrValue(rhs) as Any?
+  switch (cleanLhs, cleanRhs) {
+  case (nil, nil):
+    return true
+
+  case (nil, _), (_, nil):
+    return false
+
+  case is (Void, Void):
+    return true
+
+  case let (cleanLhsHashable, cleanRhsHashable) as (AnyHashable, AnyHashable):
+    return cleanLhsHashable == cleanRhsHashable
+
+  case let (cleanLhsArray, cleanRhsArray) as ([Any?], [Any?]):
+    guard cleanLhsArray.count == cleanRhsArray.count else { return false }
+    for (index, element) in cleanLhsArray.enumerated() {
+      if !deepEqualsEventChannelMessages(element, cleanRhsArray[index]) {
+        return false
+      }
+    }
+    return true
+
+  case let (cleanLhsDictionary, cleanRhsDictionary) as ([AnyHashable: Any?], [AnyHashable: Any?]):
+    guard cleanLhsDictionary.count == cleanRhsDictionary.count else { return false }
+    for (key, cleanLhsValue) in cleanLhsDictionary {
+      guard let cleanRhsValue = cleanRhsDictionary[key] else { return false }
+      if !deepEqualsEventChannelMessages(cleanLhsValue, cleanRhsValue) {
+        return false
+      }
+    }
+    return true
+
```

Comment:
can you also check the case when both conform to `Equatable`, then use `==` to compare them? (e.g. String is Equatable)

Reply:
String is Hashable, therefore already covered. Any type that is supported by pigeon is covered by hashable except lists and maps. And certain classes in proxy apis, but this isn't meant to solve for non-Hashable classes, as there is no real way to compare if they aren't Equitable.

Reply:
Sorry a String isn't a good example. Worth putting a comment there to explain a bit, since this is missing Non Hashable but Equatable case.